build.gradle

```diff
@@ -669,7 +674,105 @@ configure(updateBuildScript) {
     description = 'Updates the build script to the latest version'
 }
 
-// Deobfuscation
+// Parameter Deobfuscation
+
+task deobfParams {
+    doLast {
+        checkPropertyExists("channel")
+        checkPropertyExists("mappingsVersion")
+
+        String mcpDir = System.getProperty("user.home") + "/.gradle/caches/minecraft/de/oceanlabs/mcp/mcp_" + channel + "/" + mappingsVersion
+        String mcpZIP = mcpDir  + "/mcp_" + channel + "-" + mappingsVersion + "-" + minecraftVersion + ".zip"
+        String paramsCSV = mcpDir  + "/params.csv"
+
+        download.run {
+            src 'https://maven.minecraftforge.net/de/oceanlabs/mcp/mcp_' + channel + '/' + mappingsVersion + '-' + minecraftVersion + '/mcp_' + channel + '-' + mappingsVersion + '-' + minecraftVersion + '.zip'
+            dest mcpZIP
+            overwrite false
+        }
+
+        if(!file(paramsCSV).exists()) {
+            println("Extracting MCP archive ...")
+            unzip(mcpZIP, mcpDir)
+        }
+
+        println("Parsing params.csv ...")
+        Map<String, String> params = new HashMap<>()
+        Files.lines(Paths.get(paramsCSV)).forEach{line -> 
+            String[] cells = line.split(",")
+            if(cells.length > 2 && cells[0].matches("p_i?\\d+_\\d+_")) {
+                params.put(cells[0], cells[1])
+            }
+        }
+
+        out.style(Style.Success).println("Modified " + replaceParams(file("$projectDir/src/main/java"), params) + " files!")
+        out.style(Style.Failure).println("Don't forget to verify that the code still works as before!\n It could be broken due to duplicate variables existing now\n or parameters taking priority over other variables.")
+    }
+}
+
+static int replaceParams(File file, Map<String, String> params) {
+    int fileCount = 0
+
+    if(file.isDirectory()) {
+        for(File f : file.listFiles()) {
+            fileCount += replaceParams(f, params)
+        }
+        return fileCount
+    }
+    println("Visiting " + file.getName() + " ...")
+    try {
+        String content = new String(Files.readAllBytes(file.toPath()))
+        int hash = content.hashCode()
+        params.forEach{key, value ->
+            content = content.replaceAll(key, value)
+        }
+        if(hash != content.hashCode()) {
+            Files.write(file.toPath(), content.getBytes("UTF-8"))
+            return 1
+        }
+    } catch(Exception e) {
+        e.printStackTrace()
+    }
+    return 0
+}
+
+// Credit: bitsnaps (https://gist.github.com/bitsnaps/00947f2dce66f4bbdabc67d7e7b33681)
+static unzip(String zipFileName, String outputDir) {
+    byte[] buffer = new byte[1024]
```

Comment:
minor point: use at least 4k, preferably 16k buffers for i/o to better match native disk sector sizes (4k for modern ssds and advanced format hdds), it can have a significant performance impact